core/scale/collection.hpp

```diff
@@ -27,7 +27,7 @@ namespace kagome::scale::collection {
     OUTCOME_TRY(compact::encodeInteger(collection.size(), encoded_collection));
 
     TypeEncoder<T> encoder{};
-    for (size_t i = 0; i < collection.size(); ++i) {
+    for (auto i = 0; i < collection.size(); ++i) {
```

Comment:
actually size_t was better. Here i became int, while we should iterate using size_t

Reply:
+